src/helpers/serviceSelector.ts

```diff
@@ -70,16 +94,23 @@ export default class ServiceSelector {
     // Implement source event based descision logic for selecting service
   }
 
-  public static getDestinationServiceByName(name: string): DestinationService {
-    switch (name) {
-      case INTEGRATION_SERVICE.CDK_V1_DEST:
-        return this.fetchCachedService(INTEGRATION_SERVICE.CDK_V1_DEST);
-      case INTEGRATION_SERVICE.CDK_V2_DEST:
-        return this.fetchCachedService(INTEGRATION_SERVICE.CDK_V2_DEST);
-      case INTEGRATION_SERVICE.NATIVE_DEST:
-        return this.fetchCachedService(INTEGRATION_SERVICE.NATIVE_DEST);
-      default:
-        throw new PlatformError('Invalid Service');
+  public static getDestinationService(
+    events: ProcessorTransformationRequest[] | RouterTransformationRequestData[],
+  ): DestinationService {
+    const destinationDefinitionConfig: Object = events[0].destination.DestinationDefinition.Config;
+    const primaryService = this.getPrimaryDestinationService(events);
+    if (this.isComparatorEnabled(destinationDefinitionConfig)) {
+      if (this.serviceMap.has(INTEGRATION_SERVICE.COMPARATOR)) {
+        return this.serviceMap.get(INTEGRATION_SERVICE.COMPARATOR);
+      }
+      const secondaryServiceName = this.getSecondaryServiceName(destinationDefinitionConfig);
+      const secondaryService = this.getDestinationServiceByName(secondaryServiceName);
+      const comparatorService = new ComparatorService(primaryService, secondaryService);
+      this.serviceMap.set(INTEGRATION_SERVICE.COMPARATOR, comparatorService);
```

Comment:
Maybe we shouldn't cache COMPARATOR as the combination can vary right?

Reply:
Ok destination level config right yes

Reply:
Added `destination_definition_id` as part of the `Key`